### PR TITLE
feat: trace querier->ingester, stopping at gRPC layer

### PR DESCRIPTION
**Pull request by crepererum**
This adds tracing of querier->ingester request up to the point where we
perform the network request, i.e. the trace will only appear on the
querier side. We may extend this at some point to carry the tracing
information to the ingester as well.

Ref #5129.

### CODE REVIEW

File: querier/src/ingester/mod.rs [hunk 1]
```diff
@@ -791,13 +803,18 @@ impl IngesterConnection for IngesterConnectionImpl {
             .map(move |ingester_address| measured_ingester_request(ingester_address))
             .collect::<FuturesUnordered<_>>()
             .try_collect::<Vec<_>>()
-            .await?
+            .await
+            .map_err(|e| {
```
Comment (alamb) [inline querier/src/ingester/mod.rs:807]: 🤔  I wonder if `OberveIngesterRequest::error` should also be called here

**Review by alamb — APPROVED** [review]

File: querier/src/ingester/mod.rs (hunk 1 above)
Comment (crepererum) [reply querier/src/ingester/mod.rs:807]: Wrong scope, there are two recorders/spans here:

1. the overall scope that spans ALL ingester requests (see start of the method)
2. the per-ingester scope (created within `measured_ingester_request`)

File: querier/src/ingester/mod.rs [hunk 2]
```diff
@@ -739,12 +744,19 @@ impl IngesterConnection for IngesterConnectionImpl {
             // INFO: create the measurement structure outside of the async block so cancellation is
             // always measured
             let measure_me = ObserveIngesterRequest::new(request.clone(), metrics);
+            let mut span_recorder = span_recorder.child("flight request");
```
[the remark this answers is not in the document]
Comment (crepererum) [reply querier/src/ingester/mod.rs]: done

**Pull request merged**